### ISSUE TITLE
error installing flex 2.5.39 on a clean system

### ISSUE THREAD

**Issue by sguera** [hpcugent/easybuild-easyconfigs#2384]
Hello,

trying to install a GCC-4.9.3-2.25.eb on a clean system, we ended up installing flex-2.5.39. At configuration phase, flex checks for M4. We listed M4 as dependency, but at the time flex looks for it (configure time) it has still not been loaded thus resulting in an error.

We bypassed the problem adding the following lines to flex-2.5.39.eb:

preconfigopts = "ml M4/1.4.17 && "
builddependencies = [('M4','1.4.17')]

I don't know if loading the module in the preconfigopts is the right way, anyway that allowed us to get through the installation.

So far we were able to install flex through this trick. Now trying to install GCC (whole process).

**Comment by boegel** [comment]
@sguera: try this:

use

```python
toolchain = {'name': 'dummy', 'version': ''}
```

rather than

```
toolchain = {'name': 'dummy', 'version': ''dummy}
```

With the latter, any listed dependencies will indeed not be loaded...

That's a weird quirk in the framework that we need to get rid of, preferably by getting rid of the `dummy` toolchain entirely (cfr. https://github.com/hpcugent/easybuild-framework/pull/1233)

**Comment by sguera** [comment]
Hello @boegel,

we solved using

`toolchain = {'name': 'dummy', 'version': ''}`

and adding

`builddependencies = [('M4','1.4.17')]`

to flex-2.5.39.eb and flex-2.5.39-GCC-4.9.3-2.25.eb, since we were trying to install GCC-4.9.3-2.25.eb (on a clean system without M4 available). We also realized that M4 is needed as a builddep of binutils (not present so far in the easyconfigs)

**Comment by boegel** [comment]
solved with #2528 and #2681
